dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/BitstreamAccessStatusLinkRepository.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+
+package org.dspace.app.rest.repository;
+
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.UUID;
+
+import jakarta.annotation.Nullable;
+import jakarta.servlet.http.HttpServletRequest;
+import org.dspace.access.status.DefaultAccessStatusHelper;
+import org.dspace.access.status.service.AccessStatusService;
+import org.dspace.app.rest.model.AccessStatusRest;
+import org.dspace.app.rest.model.BitstreamRest;
+import org.dspace.app.rest.projection.Projection;
+import org.dspace.content.Bitstream;
+import org.dspace.content.service.BitstreamService;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Component;
+
+/**
+ * Link repository for calculating the access status of a Bitstream,
+ * including the embargo date
+ */
+@Component(BitstreamRest.CATEGORY + "." + BitstreamRest.PLURAL_NAME + "." + BitstreamRest.ACCESS_STATUS)
+public class BitstreamAccessStatusLinkRepository extends AbstractDSpaceRestRepository
+    implements LinkRestRepository {
+
+    @Autowired
+    BitstreamService bitstreamService;
+
+    @Autowired
+    AccessStatusService accessStatusService;
+
+    @PreAuthorize("hasPermission(#bitstreamId, 'BITSTREAM', 'METADATA_READ')")
+    public AccessStatusRest getAccessStatus(@Nullable HttpServletRequest request,
+                                            UUID bitstreamId,
+                                            @Nullable Pageable optionalPageable,
+                                            Projection projection) {
+        try {
+            Context context = obtainContext();
+            Bitstream bitstream = bitstreamService.find(context, bitstreamId);
+            if (bitstream == null) {
+                throw new ResourceNotFoundException("No such bitstream: " + bitstreamId);
+            }
+            AccessStatusRest accessStatusRest = new AccessStatusRest();
+            Date availabilityDate = accessStatusService.getAvailabilityDateFromBitstream(context, bitstream);
+            String status = accessStatusService.getAccessStatusFromAvailabilityDate(availabilityDate);
+            if (status == DefaultAccessStatusHelper.EMBARGO) {
+                String embargoDate = availabilityDate.toString();
+                accessStatusRest.setEmbargoDate(embargoDate);
```

Comment:
So `embargo` access condition RPs are now covered, but what about `lease` RPs?

Since the feature is agnostic to the `rpname` and thus the `AccessConditionOption` configuration it could be interesting to not call the status "embargo" in the first place, but model everything as "restricted", with or without start/end dates.

Straight-up embargo is by far the most common case though, so I'm ok if we deal with this in a separate issue after this PR.

Reply:
I agree it can be modeled in a more generic way, but it would have impacts on other features, for xoai for instance.

I also think that should be part of another issue, as I won't have the time to do it myself.